### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by ianatesfa**

### CODE REVIEW

File: src/SFA.DAS.FindEmploymentSchemes.UnitTests/Program.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿
```
Comment (lazcool) [inline src/SFA.DAS.FindEmploymentSchemes.UnitTests/Program.cs:1]: Is this file needed?

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/ErrorViewModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿
+using SFA.DAS.FindEmploymentSchemes.Web.Models;
+using Xunit;
+
+
+namespace SFA.DAS.FindEmploymentSchemes.IntegrationTests.Web.Models
+{
+    public class ErrorViewModelTests
```
Comment (lazcool) [inline src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/ErrorViewModelTests.cs:8]: Could this be moved into the unit test project?

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeDetailsModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Linq;
+using AutoFixture;
+using AutoFixture.Kernel;
+using Microsoft.AspNetCore.Html;
+using SFA.DAS.FindEmploymentSchemes.Web.Models;
+using Xunit;
+
+namespace SFA.DAS.FindEmploymentSchemes.IntegrationTests.Web.Models
+{
+    public class SchemeDetailsModelTests
```
Comment (lazcool) [inline src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeDetailsModelTests.cs:10]: Think this belongs with the unit tests too.

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeFilterViewModelTests.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+using SFA.DAS.FindEmploymentSchemes.Web.ViewModels;
+
+
+namespace SFA.DAS.FindEmploymentSchemes.IntegrationTests.Web.ViewModels
+{
+    public class SchemeFilterViewModelTests
```
Comment (lazcool) [inline src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeFilterViewModelTests.cs:11]: A unit test?

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeTests.cs [hunk 5]
```diff
@@ -0,0 +1,58 @@
+﻿using Microsoft.AspNetCore.Html;
+using SFA.DAS.FindEmploymentSchemes.Web.Models;
+using Xunit;
+
+namespace SFA.DAS.FindEmploymentSchemes.IntegrationTests.Web.Models
+{
+    public class SchemeTests
```
Comment (lazcool) [inline src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeTests.cs:7]: Unit test?

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/ErrorViewModelTests.cs (hunk 2 above)
Comment (lazcool) [reply src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/ErrorViewModelTests.cs:8]: This is already in the unit test project. Don't think we need it here also.

File: src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Controllers/PagesControllerTests.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+using Microsoft.Extensions.Logging;
+using FakeItEasy;
+using Xunit;
+using SFA.DAS.FindEmploymentSchemes.Web.Controllers;
+using SFA.DAS.FindEmploymentSchemes.Web.Models;
+
+
+namespace SFA.DAS.FindEmploymentSchemes.UnitTests.Web.Controllers
+{
+    public class PagesControllerTests
```
Comment (lazcool) [inline src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Controllers/PagesControllerTests.cs:14]: It'd be good if we tested a negative case too, i.e. a Not Found response.

File: src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Services/FilterServiceTests.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+using SFA.DAS.FindEmploymentSchemes.Web.ViewModels;
+
+
+namespace SFA.DAS.FindEmploymentSchemes.UnitTests.Web.Services
+{
+    public class FilterServiceTests
+    {
+        [Theory]
+        [ClassData(typeof(FilterServiceTestData))]
+        public void ApplyFilters_Result(IEnumerable<Scheme> expectedSchemes, SchemeFilterViewModel filters)
```
Comment (lazcool) [inline src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Services/FilterServiceTests.cs:20]: What code is this test testing?

File: src/SFA.DAS.FindEmploymentSchemes.UnitTests/Program.cs (hunk 1 above)
Comment (ianatesfa) [reply src/SFA.DAS.FindEmploymentSchemes.UnitTests/Program.cs]: Yes, it's left over from integration tests - I've removed it

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/ErrorViewModelTests.cs (hunk 2 above)
Comment (ianatesfa) [reply src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/ErrorViewModelTests.cs]: As above, I've removed this too

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeDetailsModelTests.cs (hunk 3 above)
Comment (ianatesfa) [reply src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeDetailsModelTests.cs]: Removed

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeFilterViewModelTests.cs (hunk 4 above)
Comment (ianatesfa) [reply src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeFilterViewModelTests.cs]: Removed

File: src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeTests.cs (hunk 5 above)
Comment (ianatesfa) [reply src/SFA.DAS.FindEmployerSchemes.IntegrationTests/Web/Models/SchemeTests.cs]: Removed

File: src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Controllers/PagesControllerTests.cs (hunk 6 above)
Comment (ianatesfa) [reply src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Controllers/PagesControllerTests.cs:14]: Added a couple of Not Found tests

File: src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Services/FilterServiceTests.cs (hunk 7 above)
Comment (ianatesfa) [reply src/SFA.DAS.FindEmploymentSchemes.UnitTests/Web/Services/FilterServiceTests.cs]: Removed

**Review by lazcool — APPROVED** [review]
